### PR TITLE
colmap 3.8 (new formula)

### PR DESCRIPTION
**Pull request by rgov**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

"[COLMAP](https://colmap.github.io/index.html) is a general-purpose Structure-from-Motion (SfM) and Multi-View Stereo (MVS) pipeline with a graphical and command-line interface. It offers a wide range of features for reconstruction of ordered and unordered image collections."

The software is [widely cited](https://scholar.google.com/scholar?cites=5556310561359757303&as_sdt=5,28&sciodt=0,28&hl=en).

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  CMake Error at CMakeLists.txt:91 (find_package):
    By not providing "FindCeres.cmake" in CMAKE_MODULE_PATH this project has
    asked CMake to find a package configuration file provided by "Ceres", but
    CMake did not find one.
```

**Comment by rgov** [comment]
@chenrui333 Hm, can you give any more info about your system? I do not see that output (or any output at all) when building, and the tool runs successfully for me.

**Comment by SMillerDev** [comment]
> Is this from a Linuxbrew build? Are formulae required to support Linux now?

Yes, this is from CI on Linux. Since Homebrew has first class support for linux too there should be a good reason not to support linux.

**Comment by rgov** [comment]
I'm not really sure. On macOS, CMake is able to find `ConfigCeres.cmake` from the `ceres-solver` keg no problem. On Linux it does not, yet `brew test ceres-solver` does.

On Linux, if I manually add `-DCeres_DIR=`, it does load `ConfigCeres.cmake`, but it fails to find CXSparse, and so on. I don't really know how to chase this down and running Linuxbrew under Docker is painfully slow.

**Comment by rgov** [comment]
I cannot resolve the build failure on Linux on my own, so I have marked the formula macOS only for now.

It seems that the CMake module search paths are set up fine on macOS, so I don't know if it's a problem like `std_cmake_args` is not being populated correctly on Linux.

File: Formula/colmap.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  def install
+    ENV.append "CMAKE_PREFIX_PATH", Formula["qt@5"].prefix
+
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    system "#{bin}/colmap", "-h"
```
Comment (chenrui333) [inline Formula/colmap.rb:34]: is there any way that we can do a concrete functional test?

File: Formula/colmap.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  depends_on "boost"
+  depends_on "ceres-solver"
+  depends_on "cgal"
+  depends_on "eigen"
+  depends_on "flann"
+  depends_on "freeimage"
+  depends_on "gflags"
+  depends_on "glew"
+  depends_on "glog"
+  depends_on :macos
```
Comment (chenrui333) [inline Formula/colmap.rb:18]: maybe add a code comment here for referencing the upstream issue.

File: Formula/colmap.rb (hunk 1 above)
Comment (rgov) [reply Formula/colmap.rb:34]: At the moment I don't have a good test we could run. The test would have to download a dataset of images and process them. Is there precedent for doing something like that?

**Comment by chenrui333** [comment]
@rgov if you see the other PR, it looks like the linux build actually passed.

**Comment by chenrui333** [comment]
also cc @njwhite in this thread. :)

**Comment by rgov** [comment]
The build error on Linux is related to this:

```rb
ENV.append "CMAKE_PREFIX_PATH", Formula["qt@5"].prefix
```

I'll have to investigate what's going on.

I've merged some other changes from @njwhite's formula. Theirs doesn't use QT so that's why that build didn't fail.

**Comment by chenrui333** [comment]
#141878

**Pull request closed without merging**